code/datums/components/supermatter_crystal.dm

```diff
@@ -218,11 +220,33 @@
 /datum/component/supermatter_crystal/proc/intercept_z_fall(datum/source, list/falling_movables, levels)
 	SIGNAL_HANDLER
 	for(var/atom/movable/hit_object as anything in falling_movables)
-		if(hit_object == source)
-			continue
+		if(parent == hit_object)
+			return
+
 		bumped_hit(parent, hit_object)
 	return FALL_INTERCEPTED | FALL_NO_MESSAGE
 
+/datum/component/supermatter_crystal/proc/on_z_impact(datum/source, turf/impacted_turf, levels)
+	SIGNAL_HANDLER
+
+	for(var/atom/movable/hit_object as anything in impacted_turf)
+		if(parent == hit_object)
+			return
+
+		var/atom/atom_source = source
+
+		for(var/mob/living/poor_target in impacted_turf)
```

Comment:
You seemed to have misunderstood something

Delete all of this
```dm
for(var/atom/movable/hit_object as anything in impacted_turf)
		if(parent == hit_object)
			return

		var/atom/atom_source = source
```
Right now you're iterating over every atom in the turf, just so you can iterate over ever mob in the turf for every atom there is

Just have
```dm
	for(var/mob/living/poor_target in impacted_turf)
		...
```